dataprofiler/profilers/float_column_profile.py

```diff
@@ -106,16 +106,26 @@ def _get_float_precision(cls, df_series_clean):
         :return: string representing its precision print format
         :rtype: int
         """
-        if not len(df_series_clean):
-            return 0
+        if not len(df_series_clean): return 0
 
         # Lead zeros: ^[+-.0\s]+ End zeros: \.?0+(\s|$)
         # Scientific Notation: (?<=[e])(.*) Any non-digits: \D
         r = re.compile(r'^[+-.0\s]+|\.?0+(\s|$)|(?<=[e])(.*)|\D')
-        
-        return float(df_series_clean.replace(
-            to_replace=r, value='').map(len).min())
-        
+
+        # Take a sampling of the dataset. If small use full dataset,
+        # OR 20k samples or 5% of the dataset which ever is larger.
+        sample_size = min(len(df_series_clean),
+                          max(20000, int(len(df_series_clean)/20)))
+
+        # length of sampled cells after all punctuation removed
+        len_per_float = df_series_clean.sample(sample_size).replace(
+            to_replace=r, value='').map(len)
+
+        # Determine the max and min precision
+        min_precision = float(len_per_float.min())
+        max_precision = float(len_per_float.max())
```

Comment:
What is the point of this calculation if it isn't stored or returned?

Reply:
Right now no reason (it's insanely fast). There's an additional issue up #151 which discusses returning `max_precision` and `min_precision`

Reply:
Alright then